### PR TITLE
1866: Implementing New Day MRMS Option

### LINKED ISSUE #1866: RFE:  Add "Mass Repair" button to Command Center tab
As a player, I would like to have the Mass Repair function within the Command Center, because I want to keep my most used actions within the same Command Center tab as much as possible.

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #1866 by adding an option for MRMS to trigger on New Day.

To Review: Only the commit titled "Implementing MRMS option on new day" is not refactoring, but this required a major refactor to make this possible, namely converting MassRepairService to rely on Campaign instead of CampaignGUI.

### CODE REVIEW

File: MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java [hunk 1]
```diff
@@ -634,628 +612,577 @@ public void actionPerformed(ActionEvent e) {
...
-		if (name.equals("massRepairItemPod") && !isModeWarehouse()) {
+        return pnlOptions;
+    }
+
+    private MassRepairOptionControl createMassRepairOptionControls(int type, String text,String tooltipText,
+                                                                   String activeBoxName, JPanel pnlItems,
+                                                                   int rowIdx) {
+        MassRepairOption mro = null;
+
+        for (MassRepairOption massRepairOption : campaignOptions.getMassRepairOptions()) {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java:623]: Isn't this going to throw an exception of getMassRepairOptions returns null?

**Review by NickAragua — COMMENTED** [review]
One question about a removed null check.

File: MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java:623]: Fixed it on one end, and I think I forgot to fix it on the other end... give me five and I'll double check

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/MassRepairSalvageDialog.java:623]: Yep, forgot to fix it on the other end... That's now fixed

**Review by NickAragua — APPROVED** [review]
Looking good.

**Pull request merged**